samples/bluetooth/peripheral_ancs_client/src/main.c

```diff
@@ -42,7 +49,9 @@ static const struct bt_data ad[] = {
 
 static struct bt_ancs_client ancs_c;
 
-static bool has_ancs;
+static struct bt_gatts_client gatts_c;
+
+static atomic_t discovery_flags;
```

Comment:
The variable is modified by both Discovery Manager and Service Changed indication.

Use atomic because it can be called in ISR. I am also not sure. I did not study the Bluetooth stack code path in depth.

To be honest, I did not encounter the scenario that mandates using atomic. The iPhone swiftly emitted Service Changed indications that the ANCS was always found.

It is difficult to catch race conditions. I am playing safe.